### PR TITLE
[AE] MainMenu class refactoring, MainMenuCams enum

### PR DESCRIPTION
**Pull request by pryon**
Some Update function return values are still in int format (0xFFFF0002, etc.).
As we've discussed, those will be converted to a struct, but that will have to be in another PR as this one has already gotten too big

### CODE REVIEW

File: Source/AliveLibAE/MainMenu.cpp [hunk 1]
```diff
@@ -2408,39 +2435,80 @@ signed int MainMenuController::AbeMotions_Update_4D1F50(DWORD input)
...
+    return MainMenuCams::eNoChange;
+}
 
-signed int MainMenuController::tsub_4D48C0(DWORD /*input*/)
+signed int MainMenuController::PSX_Gamemode_Selection_Update_4D48C0(DWORD input)
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    __int16 result; // ax
+    BitField32<Flags> menuFlags = this->field_23C_T80;
```
Comment (paulsapps) [inline Source/AliveLibAE/MainMenu.cpp:2474]: you can use field_23C_T80 directly instead of a local

File: Source/AliveLibAE/MainMenu.cpp [hunk 2]
```diff
@@ -2408,39 +2435,80 @@ signed int MainMenuController::AbeMotions_Update_4D1F50(DWORD input)
...
     }
 }
 
-signed int MainMenuController::tsub_Input_Check_4D49B0(DWORD /*input*/)
+signed int MainMenuController::PSX_Cooperative_Mode_Update_4D49B0(DWORD /*input*/)
 {
-    NOT_IMPLEMENTED();
-    return 0;
-}
+    DWORD held = sInputObject_5BD4E0.field_0_pads[0].field_C_held;
```
Comment (paulsapps) [inline Source/AliveLibAE/MainMenu.cpp:2462]: can be const

File: Source/AliveLibAE/MainMenu.cpp [hunk 3]
```diff
@@ -2408,39 +2435,80 @@ signed int MainMenuController::AbeMotions_Update_4D1F50(DWORD input)
...
-    NOT_IMPLEMENTED();
-    return 0;
+    __int16 result; // ax
+    BitField32<Flags> menuFlags = this->field_23C_T80;
+
+    if (input & InputCommands::eUnPause_OrConfirm)
+    {
+        sGameStartedFrame_5C1B88 = sGnFrame_5C1B84;
+        sCurrentControllerIndex_5C1BBE = 0;
+        BOOL twoPlayerModeSelected = this->field_1FC_button_index == 1;
```
Comment (paulsapps) [inline Source/AliveLibAE/MainMenu.cpp:2480]: can be const and this-> is redundant

File: Source/AliveLibAE/MainMenu.cpp [hunk 4]
```diff
@@ -2408,39 +2435,80 @@ signed int MainMenuController::AbeMotions_Update_4D1F50(DWORD input)
...
 
+    return MainMenuCams::eNoChange;
+}
 
-signed int MainMenuController::tsub_4D48C0(DWORD /*input*/)
+signed int MainMenuController::PSX_Gamemode_Selection_Update_4D48C0(DWORD input)
 {
-    NOT_IMPLEMENTED();
-    return 0;
+    __int16 result; // ax
```
Comment (paulsapps) [inline Source/AliveLibAE/MainMenu.cpp:2473]: you should try to remove this var if possible, looks like you could just use early/in place returns?

File: Source/AliveLibAE/MainMenu.cpp [hunk 5]
```diff
@@ -3001,33 +3074,35 @@ void MainMenuController::HandleMainMenuUpdate()
         }
 
         // Todo: change all field_10_fn_update function types to unsigned.
-        const unsigned int pageUpdateRet = (this->*(pPage->field_10_fn_update))(inputHeld);
+        const unsigned int pageUpdateReturnedCam = (this->*(pPage->field_10_fn_update))(inputHeld);
```
Comment (paulsapps) [inline Source/AliveLibAE/MainMenu.cpp:3077]: you could probably change these into a struct, something like:

struct NextPage
{
BYTE targetCamera;
BYTE screenChangeEffect
};

?

Then you won't need the >> 16 & 0xFF and the BYTE1() on the return value

**Comment by paulsapps** [comment]
ah I just noticed you said you'd do the struct stuff in another PR

File: Source/AliveLibAE/MainMenu.cpp (hunk 5 above)
Comment (pryon) [reply Source/AliveLibAE/MainMenu.cpp:3077]: yup that will be in the next PR, along with the function and lambda return values

**Pull request merged**